worlds/sc2/rules.py

```diff
@@ -338,6 +339,68 @@ def all_in_requirement(self, state: CollectionState):
                 and state.has_any({item_names.HIVE_MIND_EMULATOR, item_names.PSI_DISRUPTER, item_names.MISSILE_TURRET}, self.player)
 
     # HotS
+
+    def zerg_defense_rating(self, state: CollectionState, zerg_enemy: bool, air_enemy: bool = True) -> int:
+        """
+        Ability to handle defensive missions
+        :param state:
+        :param zerg_enemy:
+        :param air_enemy:
+        :return:
+        """
+        defense_score = sum((zvx_defense_ratings[item] for item in zvx_defense_ratings if state.has(item, self.player)))
+        # Twin Drones
+        if state.has(item_names.TWIN_DRONES, self.player):
+            if state.has(item_names.SPINE_CRAWLER, self.player):
+                defense_score += 1
+            if state.has(item_names.SPORE_CRAWLER, self.player) and air_enemy:
+                defense_score += 1
+        # Impaler
+        if self.morph_impaler(state):
+            defense_score += 3
+            if state.has(item_names.IMPALER_HARDENED_TENTACLE_SPINES, self.player):
+                defense_score += 1
+            if zerg_enemy:
+                defense_score += -1
+        # Lurker
+        if self.morph_lurker(state):
+            defense_score += 1
+            if state.has(item_names.LURKER_SEISMIC_SPINES, self.player):
+                defense_score += 2
+            if state.has(item_names.LURKER_ADAPTED_SPINES, self.player) and not zerg_enemy:
+                defense_score += 1
+            if zerg_enemy:
+                defense_score += 1
+        # Brood Lord
+        if self.morph_brood_lord(state):
+            defense_score += 2
+        # Corpser Roach
+        if state.has_all({item_names.ROACH, item_names.ROACH_CORPSER_STRAIN}, self.player):
+            defense_score += 1
+            if zerg_enemy:
+                defense_score += 1
+        # Igniter
+        if state.has(item_names.ROACH_PRIMAL_IGNITER_ASPECT, self.player) and (state.has(item_names.ROACH, self.player) or self.morphling_enabled) and zerg_enemy:
+            defense_score += 2
+            if state.has(item_names.PRIMAL_IGNITER_CONCENTRATED_FIRE, self.player):
+                defense_score += 1
+        # Creep Tumors
+        if state.has_any({item_names.SWARM_QUEEN, item_names.OVERLORD_OVERSEER_ASPECT}, self.player):
+            defense_score += 1
+            if state.has(item_names.MALIGNANT_CREEP, self.player):
+                defense_score += 1
```

Comment:
Not Zerg enemy? While defending you don't want to make the enemy faster

Reply:
True, I'll fix that